avro/src/test/java/com/bakdata/kafka/AvroDeadLetterTransformerTest.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.bakdata.kafka;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KeyValueMapper;
+import org.apache.kafka.streams.kstream.Produced;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.jooq.lambda.Seq;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@ExtendWith(SoftAssertionsExtension.class)
+class AvroDeadLetterTransformerTest extends ErrorCaptureTopologyTest {
+    @InjectSoftAssertions
+    private SoftAssertions softly;
+    private static final String ERROR_TOPIC = "errors";
+    private static final String OUTPUT_TOPIC = "output";
+    private static final String INPUT_TOPIC = "input";
+    private static final Serde<String> STRING_SERDE = Serdes.String();
+    public static final String DEADLETTER_DESCRIPTION = "Description";
+    public static final String ERROR_MESSAGE = "ERROR!";
+
+    @Mock
+    KeyValueMapper<Integer, String, KeyValue<Integer, String>> mapper;
+
+    @Override
+    protected void buildTopology(final StreamsBuilder builder) {
+        final KStream<Integer, String> input = builder.stream(INPUT_TOPIC, Consumed.with(null, STRING_SERDE));
+        final KStream<Integer, ProcessedKeyValue<Integer, String, String>> mapped =
+                input.map(ErrorCapturingKeyValueMapper.captureErrors(this.mapper));
+        mapped.flatMapValues(ProcessedKeyValue::getValues)
+                .to(OUTPUT_TOPIC, Produced.valueSerde(STRING_SERDE));
+        mapped.flatMap(ProcessedKeyValue::getErrors)
+                .transformValues(DeadLetterTransformer.create(DEADLETTER_DESCRIPTION, new AvroDeadLetterConverter()))
```

Comment:
@torbsto What do you think about making this a bit more readable with sth like `DeadLetterTransformer.transform(toAvro(), DEADLETTER_DESCRIPTION)` where toAvro is just a static method in AvroDeadLetterConverter? Even more advanced would be sth like `describedAs(DESCRIPTION).transformedTo(avro())`